### PR TITLE
Add Free Time feature to model

### PR DESCRIPTION
**Pull request by andrefoo**
TimetaBRO needs to know a person's free time to identify
 when they are available as lunch buddies.

 Let's add a FreeTime class in the model.
 For storage purposes, add JsonAdaptedFreeTime and
 adapt Person and JsonAdaptedPerson classes to compose of
  a Set of free times.

 A few cascading errors have been addressed, adding
 freetime support to PersonBuilders and Data and Util classes.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -21,11 +21,7 @@
 import seedu.address.logic.Messages;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
-import seedu.address.model.person.Address;
-import seedu.address.model.person.Email;
-import seedu.address.model.person.Name;
-import seedu.address.model.person.Person;
-import seedu.address.model.person.Phone;
+import seedu.address.model.person.*;
```
Comment (andrefoo) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:24]: don't do this

Comment (owenyeo) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:24]: Yes

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 2]
```diff
@@ -6,11 +6,7 @@
 
 import seedu.address.model.AddressBook;
 import seedu.address.model.ReadOnlyAddressBook;
-import seedu.address.model.person.Address;
-import seedu.address.model.person.Email;
-import seedu.address.model.person.Name;
-import seedu.address.model.person.Person;
-import seedu.address.model.person.Phone;
+import seedu.address.model.person.*;
```
Comment (owenyeo) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:9]: Another mistake here

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 3]
```diff
@@ -10,11 +10,7 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import seedu.address.commons.exceptions.IllegalValueException;
-import seedu.address.model.person.Address;
-import seedu.address.model.person.Email;
-import seedu.address.model.person.Name;
-import seedu.address.model.person.Person;
-import seedu.address.model.person.Phone;
+import seedu.address.model.person.*;
```
Comment (owenyeo) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:13]: Same as above

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 4]
```diff
@@ -2,15 +2,12 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CliSyntax.*;
```
Comment (owenyeo) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:5]: Same as above

File: src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java [hunk 5]
```diff
@@ -1,8 +1,6 @@
 package seedu.address.logic.commands;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (owenyeo) [inline src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java:3]: Same as above

File: src/test/java/seedu/address/testutil/PersonBuilder.java [hunk 6]
```diff
@@ -3,11 +3,7 @@
 import java.util.HashSet;
 import java.util.Set;
 
-import seedu.address.model.person.Address;
-import seedu.address.model.person.Email;
-import seedu.address.model.person.Name;
-import seedu.address.model.person.Person;
-import seedu.address.model.person.Phone;
+import seedu.address.model.person.*;
```
Comment (owenyeo) [inline src/test/java/seedu/address/testutil/PersonBuilder.java:6]: Same as above

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 7]
```diff
@@ -1,22 +1,14 @@
 package seedu.address.logic.parser;
 
 import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CliSyntax.*;
```
Comment (owenyeo) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:4]: Same as above

**Review by owenyeo — CHANGES_REQUESTED** [review]
Change import statements

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java (hunk 7 above)
Comment (andrefoo) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java]: okay done

**Review by owenyeo — APPROVED** [review]
LGTM

**Pull request merged**